spec/controllers/restaurants_controller_spec.rb

```diff
@@ -0,0 +1,179 @@
+require 'rails_helper'
+
+RSpec.describe RestaurantsController, type: :controller do
+  def response_body
+    JSON.parse(response.body)
+  end
+
+  def create_cuisine
+    create(:cuisine, name: 'C1')
+  end
+
+  let(:restaurant1) {create(:restaurant, name: 'R1')}
+  let(:restaurant2) {create(:restaurant, name: 'R2')}
+  let(:restaurant3) {create(:restaurant, name: 'R3')}
+  let(:restaurants) { [restaurant1, restaurant2, restaurant3]}
+  let(:rest_params) {FactoryBot.attributes_for(:restaurant)}
+  let(:rest_no_name) {{cuisine_id: 1, delivery_time: 50}}
+  let(:rest_no_cuisine) {{name: 'Rest', delivery_time: 50}}
+
+
+
+  describe 'Index' do
+
+    context 'Returns all the restaurants sorted by name' do
+      before do
+        create_cuisine
+        allow(Restaurant).to receive(:sorted_by_name_asc).and_return(restaurants)
+        get :index
+      end
+
+      it 'returns an array of 3 restaurants' do
+        expect(response_body.length).to eq(3)
+      end
+
+      it 'Returns status code 200' do
+        expect(response.status).to eq(200)
+      end
+
+      it 'Checking the first restaurant in the array' do
+        expect(response_body.first['id']).to eq(restaurant1.id)
```

Comment:
there is not much point in this test because you never actually call you `sorted_by_name_asc` method, you stub it so this will always be true. If you do want to test this `sorted_by_name_asc` scope, do it in the model and don't stub it.